opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -128,7 +128,15 @@ pub(crate) struct GenevaConfigClientConfig {
     pub(crate) namespace: String,
     pub(crate) region: String,
     pub(crate) config_major_version: u32,
-    pub(crate) auth_method: AuthMethod, // agent_identity and agent_version are hardcoded for now
+    pub(crate) auth_method: AuthMethod,
+    /// User agent for the application. Will be formatted as "<application> (RustGenevaClient/0.1)".
+    /// If None, defaults to "RustGenevaClient/0.1".
+    /// 
+    /// Examples:
+    /// - None: "RustGenevaClient/0.1"
+    /// - Some("MyApp/2.1.0"): "MyApp/2.1.0 (RustGenevaClient/0.1)"
+    /// - Some("ProductionService-1.0"): "ProductionService-1.0 (RustGenevaClient/0.1)"
+    pub(crate) user_agent_suffix: Option<&'static str>,
```

Comment:
Need to add validation for length and character set compliance. I believe it should be ASCII characters only and not have control characters and line breaks.

Reply:
Good point - have added the validations, and tests to validate if they work correctly.